My project/Assets/Scripts/CreatureLogic/Actions/Breeding/Breed.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Breed : IAction
+{
+    //class variables
+    BaseCreature target;
+    CreatureData data;
+    RangeScanner scanner;
+    Rigidbody2D rb;
+    float breedDuration;
+    float breedTimeEnd;
+    float nextBreedTime;
```

Comment:
can we simplify this by having removing nextBreedTime and having it be derived from breedTimeEnd + breedCooldown?